File: 0860-lemonade-change/0860-lemonade-change.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
        int five = 0;
        int ten = 0;
        
        for(int bill : bills) {
            if(bill == 5) five += 1;

            else if(bill == 10) {
                if(five!=0) {
                    five = five - 1;
                    ten = ten + 1;
                }
                else return false;
            }
            else if(bill == 20) {
                if(ten != 0 && five !=0) {
                    ten = ten - 1;
                    five = five -1;
                }
                else if(five >= 3) five -= 3;
                else return false;
            }
        }
        return true;
    }
};

File: 0860-lemonade-change/0860-lemonade-change.py
class Solution(object):
    def lemonadeChange(self, bills):
        """
        :type bills: List[int]
        :rtype: bool
        """
        five = ten = 0

        for bill in bills:
            if(bill == 5): five+=1

            elif (bill == 10) :
                if(five != 0) :
                    five = five - 1
                    ten = ten + 1
                else : 
                    return False

            elif (bill == 20) :
                if(five != 0 and ten != 0):
                    ten = ten - 1
                    five = five - 1
                elif(five >= 3) :
                    five = five - 3
                else :
                    return False
                    
        return True